internal/storagenode/logstream/committer.go

```diff
@@ -260,40 +259,56 @@ func (cm *committer) commitInternal(cc storage.CommitContext) (err error) {
 	}()
 
 	iter := cm.commitWaitQ.peekIterator()
-	for i := 0; i < numCommits; i++ {
-		llsn := cc.CommittedLLSNBegin + types.LLSN(i)
-		glsn := cc.CommittedGLSNBegin + types.GLSN(i)
+	numLogEntries := 0
+	numCWTs := 0
+	for numLogEntries < numCommits && iter.valid() {
+		cwt := iter.task()
 
-		if uncommttedLLSNBegin+types.LLSN(i) != llsn {
-			err = errors.New("log stream: committer: llsn mismatch")
-			return err
-		}
+		// invariant: numCommits-numLogEntries >= cwt.size
+		for i := range cwt.size {
+			llsn := cc.CommittedLLSNBegin + types.LLSN(numLogEntries)
+			glsn := cc.CommittedGLSNBegin + types.GLSN(numLogEntries)
 
-		// Since the number of tasks in commitWaitQ is inspected above, cwt
-		// must exist.
-		cwt := iter.task()
-		cwt.awg.setGLSN(glsn)
+			if uncommttedLLSNBegin+types.LLSN(numLogEntries) != llsn {
+				err = errors.New("log stream: committer: llsn mismatch")
+				return err
+			}
 
-		err = cb.Set(llsn, glsn)
-		if err != nil {
-			return err
+			if len(cwt.awgs) > 0 {
```

Comment:
It's not necessary because `len(cwt.awgs)` and `cwt.size` are always same.

Reply:
`len(cwt.awgs)` does not have to be equal to `cwt.size`. Please refer to this code.

https://github.com/kakao/varlog/blob/fdd3cd226cb173105b74665a697d247de858ae26/internal/storagenode/logstream/committer.go#L422-L433

There are some typos in the comments, but they'll be removed soon.